setup/docker/docker-compose.yml

```diff
@@ -1,47 +1,46 @@
-nginx:
-  image: nginx
-  links:
-    - enketo_express
-  ports:
-    - "80:80"
-    - "443:443"
-  env_file:
-    - ./envfile.txt
-  volumes:
-    - ./nginx/:/tmp/enketo_express_nginx/:ro
-    - ./secrets/:/tmp/enketo_express_secrets/:ro
-    - ../../config/:/srv/src/enketo_express/config/
-  command: bash /tmp/enketo_express_nginx/nginx_command.bash
-  restart: on-failure
+version: "3"
 
-enketo_express:
-  # image: kobotoolbox/enketo_express
-  build: ../..
-  links:
-    - redis_main
-    - redis_cache
-  env_file:
-    - ./envfile.txt
-  volumes:
-    - ../../config/:/srv/src/enketo_express/config
-    - ./.vols/checksum/:/srv/src/enketo_express/checksum
-    - ./.vols/build/css:/srv/src/enketo_express/public/css
-    - ./.vols/build/js:/srv/src/enketo_express/public/js/build
-    - ./.vols/build/locales:/srv/src/enketo_express/locales/build
-    - ./secrets/:/srv/src/enketo_express/setup/docker/secrets/
-  restart: on-failure
+services:
+  nginx:
+    image: nginx
+    # Define `ports` ONLY for public services, not for internal communication
+    # between containers
+    ports:
+      - "80:80"
+      - "443:443"
+    env_file:
+      - ./envfile.txt
+    volumes:
+      - ./nginx/:/tmp/enketo_express_nginx/:ro
+      - ./secrets/:/tmp/enketo_express_secrets/:ro
+      - ../../config/:/srv/src/enketo_express/config/
+    command: bash /tmp/enketo_express_nginx/nginx_command.bash
+    restart: unless-stopped
 
-redis_main:
-  image: redis:2.6
-  # Map our "main" Redis config into the container.
-  volumes:
-    - ../redis/conf/redis-enketo-main.conf:/etc/redis/redis.conf:ro
-    - ./redis_main_data/:/data/
-  restart: on-failure
+  enketo_express:
+    # image: kobotoolbox/enketo_express
+    build: ../..
+    env_file:
+      - ./envfile.txt
+    volumes:
+      - ../../config/:/srv/src/enketo_express/config
+      - ./.vols/checksum/:/srv/src/enketo_express/checksum
+      - ./.vols/build/css:/srv/src/enketo_express/public/css
+      - ./.vols/build/js:/srv/src/enketo_express/public/js/build
+      - ./.vols/build/locales:/srv/src/enketo_express/locales/build
+      - ./secrets/:/srv/src/enketo_express/setup/docker/secrets/
+    restart: unless-stopped
 
-redis_cache:
-  image: redis:2.6
-  # Map our "cache" Redis config into the container.
-  volumes:
-    - ../redis/conf/redis-enketo-cache.conf:/etc/redis/redis.conf:ro
-  restart: on-failure
+  redis_main:
+    image: redis:2.6
```

Comment:
We should use `redis:3.x` at least

Reply:
I'll go to 5. I tested it from scratch and with an existing 2.6 database; both worked fine.